### PR TITLE
Use `buildSync` and tx constants from common-sdk

### PR DESCRIPTION
**Pull request by terranmoccasin**
Use `buildSync` and also transaction constants from `@orca-so/common-sdk`
https://github.com/orca-so/orca-sdks/pull/41

- [x] Need to deploy `@orca-so/common-sdk` at `dc438328f53c999711083d2dca4b92d4602e9664` as 0.2.2-beta.1

### CODE REVIEW

File: sdk/package.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  "version": "0.10.0-beta.2",
   "description": "Typescript SDK to interact with Orca's Whirlpool program.",
   "license": "Apache-2.0",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "dependencies": {
     "@coral-xyz/anchor": "~0.27.0",
     "@metaplex-foundation/mpl-token-metadata": "1.2.5",
-    "@orca-so/common-sdk": "0.2.1",
+    "@orca-so/common-sdk": "0.2.2-beta.1",
```
Comment (yugure-orca) [inline sdk/package.json:11]: as you know, ``0.2.2`` have been published. 👍

File: sdk/src/router/public/router-utils.ts [hunk 2]
```diff
@@ -114,13 +117,13 @@ export class RouterUtils {
...
         try {
-          v0TxSize = await tx.txnSize({
+          v0TxSize = tx.txnSize({
             latestBlockhash: MEASUREMENT_BLOCKHASH,
             maxSupportedTransactionVersion: opts.maxSupportedTransactionVersion,
             lookupTableAccounts,
           });
 
-          if (v0TxSize !== undefined && v0TxSize < ENCODED_LIMIT) {
+          if (v0TxSize !== undefined && v0TxSize < TX_BASE64_ENCODED_SIZE_LIMIT) {
```
Comment (yugure-orca) [inline sdk/src/router/public/router-utils.ts:126]: As you know, the result of ``txnSize`` is now less than or equal to ``TX_SIZE_LIMIT``. 🙂

**Review by odcheung — APPROVED** [review]

File: sdk/src/router/public/router-utils.ts [hunk 3]
```diff
@@ -92,11 +94,11 @@ export class RouterUtils {
       }
 
       try {
-        const legacyTxSize = await tx.txnSize({
+        const legacyTxSize = tx.txnSize({
           latestBlockhash: MEASUREMENT_BLOCKHASH,
           maxSupportedTransactionVersion: "legacy",
         });
-        if (legacyTxSize !== undefined && legacyTxSize < TX_LIMIT) {
+        if (legacyTxSize !== undefined && legacyTxSize < TX_SIZE_LIMIT) {
```
Comment (yugure-orca) [inline sdk/src/router/public/router-utils.ts:101]:
```suggestion
        if (legacyTxSize !== undefined && legacyTxSize <= TX_SIZE_LIMIT) {
```

File: sdk/src/router/public/router-utils.ts [hunk 4]
```diff
@@ -114,13 +116,13 @@ export class RouterUtils {
...
         try {
-          v0TxSize = await tx.txnSize({
+          v0TxSize = tx.txnSize({
             latestBlockhash: MEASUREMENT_BLOCKHASH,
             maxSupportedTransactionVersion: opts.maxSupportedTransactionVersion,
             lookupTableAccounts,
           });
 
-          if (v0TxSize !== undefined && v0TxSize < ENCODED_LIMIT) {
+          if (v0TxSize !== undefined && v0TxSize < TX_SIZE_LIMIT) {
```
Comment (yugure-orca) [inline sdk/src/router/public/router-utils.ts:125]:
```suggestion
          if (v0TxSize !== undefined && v0TxSize <= TX_SIZE_LIMIT) {
```

**Review by yugure-orca — COMMENTED** [review]
ah, I was distracted by the constant, but ``<`` is ``<=``.

**Pull request merged**